spec/requests/providers/means/student_finances_controller_spec.rb

```diff
@@ -0,0 +1,217 @@
+require "rails_helper"
+
+RSpec.describe Providers::Means::StudentFinancesController do
+  let(:legal_aid_application) { create(:legal_aid_application, student_finance: nil) }
+  let(:provider) { legal_aid_application.provider }
+
+  describe "GET /providers/applications/:legal_aid_application_id/means/student_finance" do
+    subject(:request) do
+      get providers_legal_aid_application_means_student_finance_path(legal_aid_application)
+    end
+
+    it "returns ok" do
+      login_as provider
+
+      request
+
+      expect(response).to have_http_status(:ok)
+    end
+
+    context "when the application has student finance" do
+      let(:legal_aid_application) { create(:legal_aid_application, student_finance: true) }
+
+      it "checks the `Yes` radio button and displays the amount" do
+        _student_finance = create(
+          :irregular_income,
+          amount: 1234.56,
+          legal_aid_application:,
+        )
+        login_as provider
+
+        request
+
+        expect(page).to have_checked_field("Yes")
+        expect(page).to have_field("irregular_income[amount]", with: "1234.56")
+      end
+    end
+
+    context "when the application does not have student finance" do
+      let(:legal_aid_application) { create(:legal_aid_application, student_finance: false) }
+
+      it "checks the `No` radio button" do
+        login_as provider
+
+        request
+
+        expect(page).to have_checked_field("No")
+      end
+    end
+
+    context "when the provider is not authenticated" do
+      before { request }
+
+      it_behaves_like "a provider not authenticated"
+    end
+
+    context "when the provider is not authorised" do
+      before do
+        login_as create(:provider)
+        request
+      end
+
+      it_behaves_like "an authenticated provider from a different firm"
+    end
+  end
+
+  describe "PATCH /providers/applications/:legal_aid_application_id/means/student_finance" do
+    subject(:request) do
+      patch providers_legal_aid_application_means_student_finance_path(legal_aid_application),
+            params: params.merge(button_clicked)
+    end
+
+    let(:params) { { irregular_income: { student_finance:, amount: } } }
+    let(:student_finance) { "false" }
+    let(:amount) { "" }
+    let(:button_clicked) { {} }
+    let(:continue_button) { { continue_button: "Save and continue" } }
+
+    context "when the provider continues" do
+      let(:button_clicked) { continue_button }
+
+      context "when the provider selects `Yes`" do
+        let(:student_finance) { "true" }
+        let(:amount) { "5000" }
+
+        it "updates the application" do
+          login_as provider
```

Comment:
personally i'd prefer a context "when authenticated", with before { login_as provider } to remove the setup from these examples